src/interchange/src/avro.rs

```diff
@@ -1289,6 +1361,27 @@ impl DebeziumDeduplicationState {
                     }
                 } else {
                     if let Some(seen_offsets) = seen_offsets.get_mut(file) {
+                        if let Some(range) = range {
+                            if let Some(upstream_time_millis) = upstream_time_millis {
+                                if upstream_time_millis < range.pad_start {
+                                    return should_skip.is_none();
+                                }
+                                if range.pad_start < upstream_time_millis
```

Comment:
what happens if upstream_time_millis is exactly range.pad_start ? Is there an off-by-one error here?

Reply:
Possibly, I'll add some comments explaining why it doesn't matter, though.

Reply:
actually you were right, got rid of the pad_start comparison entirely.